packages/nimble-components/build/generate-icons/source/index.js

```diff
@@ -70,10 +75,21 @@ export const ${tagName} = DesignSystem.tagFor(${className});
     fileCount += 1;
 
     allIconsFileContents = allIconsFileContents.concat(`export { ${className} } from './${fileName}';\n`);
+
+    allIconsReactFileImports = allIconsReactFileImports.concat(`    ${className},\n`);
+    allIconsReactFileExports = allIconsReactFileExports.concat(`export const Nimble${className} = wrap(${className});\n`);
 }
 console.log(`Finshed writing ${fileCount} icon component files`);
 
 const allIconsFilePath = path.resolve(iconsDirectory, 'all-icons.ts');
 console.log('Writing all-icons file');
 fs.writeFileSync(allIconsFilePath, allIconsFileContents, { encoding: 'utf-8' });
 console.log('Finished writing all-icons file');
+
+allIconsReactFileImports = allIconsReactFileImports.concat('} from \'./all-icons\';\n');
+const allIconsReactFileContents = allIconsReactFileImports.concat(allIconsReactFileExports);
+
+const allIconsReactFilePath = path.resolve(iconsDirectory, 'all-icons.react.tsx');
```

Comment:
I don't think it would be a huge effort to emit these as `/icons/tests/nimble-icon.react.tsx`
That would follow our other icon patterns and prevent two issues:
- A test importing a single icon will import every icon. Each story is a support output now so importing any icon will make them all larger now (possibly)
- Modifying / adding an icon will cause every snapshot to be remade since pretty much every story will have a dependency on all-icons.

I think it's worthwhile to emit the separate icon files for consistency and avoid the (possible) downsides.

Reply:
Sounds good - we now have the icon react wrapper generated like so – 

`packages/nimble-components/src/icons/tests/add.react.tsx`

```jsx
// AUTO-GENERATED FILE - DO NOT EDIT DIRECTLY
// See generation source in nimble-components/build/generate-icons

import { wrap } from '../../utilities/tests/react-wrapper';
import { IconAdd } from '../add';

export const NimbleIconAdd = wrap(IconAdd);
```